libyul/backends/evm/EVMDialect.cpp

```diff
@@ -325,6 +316,7 @@ std::vector<std::optional<BuiltinFunctionForEVM>> createBuiltins(langutil::EVMVe
 				SideEffects::Write,  // memory
 				SideEffects::None    // transientStorage
 			},
+			ControlFlowSideEffects::fromInstruction(evmasm::Instruction::CODECOPY),
```

Comment:
I wanted to suggest to use `EVMDialect::sideEffectsOfInstruction(evmasm::Instruction::CODECOPY)` for `SideEffects` here too, but apparently the `movableApartFromEffects` flag here differs from what `SemanticInformation::movableApartFromEffects()` returns for this instruction.

The discrepancy Looks like a bug to me and confirms that we should be getting these values from `SemanticInformation` as a single source of truth instead of repeating them. Unless this case is actually intentional? @ekpyron?

Reply:
I'm only seeing this now - it shouldn't actually make a difference in this case, but yeah, we can use the ``SemanticInformation`` version instead.

Reply:
(as in, we probably should, the fact that it doesn't make a difference is more of a happy accident than intent here)

Reply:
Good. Here's a PR that fixes it: #15556.

Turns out that the flag was randomly flipped in a refactor: https://github.com/ethereum/solidity/pull/9283#discussion_r452236682. Chris actually discovered that and requested a change, and it was marked as resolved, but the fix was apparently never applied before the PR was merged.

Fortunately looks like the value of this flag does not change anything for instructions that write to memory and this is the case here, so nothing was actually broken.